src/mvesuvio/oop/AnalysisRoutine.py

```diff
@@ -0,0 +1,1066 @@
+        dE1 = np.where(spectrums < 135, 88.7, 73)  # meV, STD
+        dE1_lorz = np.where(spectrums < 135, 40.3, 24)  # meV, HFHM
+        dTOF = np.repeat(0.37, L)  # us
+        dTheta = np.repeat(0.016, L)  # rad
+        dL0 = np.repeat(0.021, L)  # meters
+        dL1 = np.repeat(0.023, L)  # meters
+
+        resolutionPars = np.vstack((dE1, dTOF, dTheta, dL0, dL1, dE1_lorz)).transpose()
+        return resolutionPars
+
+
+    def calculateKinematicsArrays(self, instrPars):
+        """Kinematics quantities calculated from TOF data"""
+
+        dataX = self._dataX
+
+        mN, Ef, en_to_vel, vf, hbar = loadConstants()
+        det, plick, angle, T0, L0, L1 = np.hsplit(instrPars, 6)  # each is of len(dataX)
+        t_us = dataX - T0  # T0 is electronic delay due to instruments
+        v0 = vf * L0 / (vf * t_us - L1)
+        E0 = np.square(
+            v0 / en_to_vel
+        )  # en_to_vel is a factor used to easily change velocity to energy and vice-versa
+
+        delta_E = E0 - Ef
+        delta_Q2 = (
+            2.0
+            * mN
+            / hbar**2
+            * (E0 + Ef - 2.0 * np.sqrt(E0 * Ef) * np.cos(angle / 180.0 * np.pi))
+        )
+        delta_Q = np.sqrt(delta_Q2)
+        return v0, E0, delta_E, delta_Q  # shape(no of spectrums, no of bins)
+
+
+    @staticmethod
+    def reshapeArrayPerSpectrum(A):
+        """
+        Exchanges the first two axes of an array A.
+        Rearranges array to match iteration per spectrum
+        """
+        return np.stack(np.split(A, len(A), axis=0), axis=2)[0]
+
+
+    def convertDataXToYSpacesForEachMass(self, dataX, delta_Q, delta_E):
+        "Calculates y spaces from TOF data, each row corresponds to one mass"
+
+        # Prepare arrays to broadcast
+        dataX = dataX[np.newaxis, :, :]
+        delta_Q = delta_Q[np.newaxis, :, :]
+        delta_E = delta_E[np.newaxis, :, :]
+
+        mN, Ef, en_to_vel, vf, hbar = loadConstants()
+        masses = self._masses.reshape(self._masses.size, 1, 1)
+
+        energyRecoil = np.square(hbar * delta_Q) / 2.0 / masses
+        ySpacesForEachMass = (
+            masses / hbar**2 / delta_Q * (delta_E - energyRecoil)
+        )  # y-scaling
+        return ySpacesForEachMass
+
+
+    def fitNcpToArray(self):
+        """Takes dataY as a 2D array and returns the 2D array best fit parameters."""
+
+        for row in range(len(self._dataY)):
+
+            specFitPars = self.fitNcpToSingleSpec(row)
+
+            self._fit_parameters[row] = specFitPars
+
+            if np.all(specFitPars == 0):
+                print("Skipped spectra.")
+            else:
+                with np.printoptions(
+                    suppress=True, precision=4, linewidth=200, threshold=sys.maxsize
+                ):
+                    print(specFitPars)
+
+        assert ~np.all(
+            self._fit_parameters == 0
+        ), "Either Fits are all zero or assignment of fitting not working"
```

Comment:
Is `~np.all(self._fit_parameters == 0)` equivalent to `np.any(self._fit_parameters)`? Maybe I'm missing a subtle distinction.